gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+
+    ZoneField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ZoneField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ZONE_ID.selector());
+      for (ZoneField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a DNS record.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code DnsService#listDnsRecords(BigInteger, DnsRecordOptions...)} or {@code
+   * DnsService#listDnsRecords(String, DnsRecordOptions...)}. The name is always returned even if
+   * not selected.
+   */
+  enum DnsRecordField {
+    DNS_RECORDS("rrdatas"),
+    NAME("name"),
+    TTL("ttl"),
+    TYPE("type");
+
+    private final String selector;
+
+    DnsRecordField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(DnsRecordField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(NAME.selector());
+      for (DnsRecordField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a change request.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code DnsService#applyChangeRequest(ChangeRequest, BigInteger, ChangeRequestFieldOptions...)}
+   * or {@code DnsService#applyChangeRequest(ChangeRequest, String, ChangeRequestFieldOptions...)}
+   * The ID is always returned even if not selected.
+   */
+  enum ChangeRequestField {
+    ID("id"),
+    START_TIME("startTime"),
+    STATUS("status"),
+    ADDITIONS("additions"),
+    DELETIONS("deletions");
+
+    private final String selector;
+
+    ChangeRequestField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ChangeRequestField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ID.selector());
+      for (ChangeRequestField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting keys for listing change requests. The only currently supported sorting key is the
+   * change sequence.
+   */
+  enum ChangeRequestSortingKey {
+    CHANGE_SEQUENCE("changeSequence");
+
+    private final String selector;
+
+    ChangeRequestSortingKey(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
```

Comment:
I know we have it as public in other APIs, but I do wonder if that is the right decision (vs package scope).